src/shared/components/oncoprint/DataUtils.ts

```diff
@@ -98,11 +98,22 @@ export function selectDisplayValue(counts:{[value:string]:number}, priority:{[va
     }
 };
 
-export function fillGeneticTrackDatum(
-    newDatum:Partial<GeneticTrackDatum>,
+type FilledGeneticTrackDatum<T> = T & Pick<GeneticTrackDatum, (
+    'gene'
+    | 'data'
+    | 'disp_fusion'
+    | 'disp_cna'
+    | 'disp_mrna'
+    | 'disp_prot'
+    | 'disp_mut'
+    | 'disp_germ'
+)>;
+export function fillGeneticTrackDatum<T>(
+    oldDatum: T,
     hugoGeneSymbol:string,
     data:AnnotatedExtendedAlteration[]
-):GeneticTrackDatum {
+): FilledGeneticTrackDatum<T> {
+    const newDatum: Partial<FilledGeneticTrackDatum<T>> = oldDatum;
```

Comment:
why are you doing this, why not just keep the parameter name and type it Partial<FilledGeneticTrackDatum<T>> instead of T

Reply:
in general I don't understand the point of what you've done here with FilledGeneticTrackDatum and the generic type T. can you explain more about that?

Reply:
I made the type of the function's interface more explicit for clarity, but that might not be very effective if understanding it requires questions 😄  What I found (and documented) that this function does is take any object and return something that has whatever properties that object had, plus several specific `GeneticTrackDatum` properties. It doesn't depend on any of the input object's properties, and can't claim to return a complete `GeneticTrackDatum` if given a partial one (in which all of the properties may be missing). It *can* claim to add its list of properties to whatever object you pass to it though.

Reply:
I think that's being too pedantic.. typescript has certain limitations of expressiveness and inference that I'm getting around by using `Partial` for the parameter, and then casting at the end.

Reply:
I was able to express it in a way that helped me understand the function without too much effort, but if you feel like it will confuse others I can leave this commit out; it's otherwise unrelated to the merged track feature, after all.

Reply:
yes please take it out, i think it adds unnecessary complexity